service_container/injection_types.rst

```diff
@@ -228,12 +235,15 @@ Another possibility is setting public fields of the class directly::
     .. code-block:: php
 
         // config/services.php
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
         use App\Mail\NewsletterManager;
-        use Symfony\Component\DependencyInjection\Reference;
 
-        // ...
-        $container->register('newsletter_manager', NewsletterManager::class)
-            ->setProperty('mailer', new Reference('mailer'));
+        return function(ContainerConfigurator $configurator) {
+            $container = $configurator->services();
+            $container->set('app.newsletter_manager, NewsletterManager::class)
+                ->property('mailer', ref('mailer'));
```

Comment:
We should maybe show both `->arg('$mailer', ref('mailer))` and this with a comment.